pom.xml

```diff
@@ -137,6 +138,12 @@
         <maven.source.plugin.version>3.0.1</maven.source.plugin.version>
         <maven.surefire.plugin.version>2.22.1</maven.surefire.plugin.version>
         <nexus.staging.maven.plugin.version>1.6.8</nexus.staging.maven.plugin.version>
+        <maven.checkstyle.plugin.version>3.0.0</maven.checkstyle.plugin.version>
+
+        <checkstyle.basedir>${main.basedir}/build/rules</checkstyle.basedir>
+        <checkstyle.config.location>${main.basedir}/build/rules/checkstyle_rules.xml</checkstyle.config.location>
```

Comment:
once you move your checkstyle rules under `checkstyle` folder, you can define this as `<checkstyle.config.location>checkstyle/checkstyle_rules.xml</checkstyle.config.location>`

Reply:
Makes sense. Tested it out locally, it works fine.